admin/admin-base.php

```diff
@@ -111,46 +111,70 @@ public function init() {
 	}
 
 	/**
-	 * Adds the link to the languages panel in the WordPress admin menu
+	 * Adds the links to Polylang's admin panels in the WordPress admin menu.
 	 *
 	 * @since 0.1
 	 *
 	 * @return void
 	 */
-	public function add_menus() {
+	public function add_menus(): void {
 		global $admin_page_hooks;
 
-		// Prepare the list of tabs
-		$tabs = array( 'lang' => __( 'Languages', 'polylang' ) );
+		$parent    = false;
+		$page_type = 'languages';
 
-		// Only if at least one language has been created
-		if ( $this->model->has_languages() ) {
-			$tabs['strings'] = __( 'Translations', 'polylang' );
-		}
+		foreach ( $this->get_sub_menu_items() as $tab => $tab_data ) {
+			$page = 'lang' === $tab ? 'mlang' : "mlang_$tab";
 
-		$tabs['settings'] = __( 'Settings', 'polylang' );
+			if ( empty( $parent ) ) {
+				$parent = $page;
+				add_menu_page( $tab_data['label'], __( 'Languages', 'polylang' ), $tab_data['capability'], $page, '__return_null', 'dashicons-translation' );
+				$admin_page_hooks[ $page ] = $page_type; // Hack to avoid the localization of the hook name. See: https://core.trac.wordpress.org/ticket/18857
+			}
 
-		/**
-		 * Filter the list of tabs in Polylang settings
+			add_submenu_page( $parent, $tab_data['label'], $tab_data['label'], $tab_data['capability'], $page, array( $this, 'languages_page' ) );
+		}
+
+		/*
+		 * Get rid of the `toplevel` prefix in hook names.
 		 *
-		 * @since 1.5.1
+		 * In the WP admin, if an admin screen is the first of its menu (like the PLL's "Languages" screen), the hooks
+		 * fired in the screen get a `toplevel` prefix (ex: `toplevel_page_mlang`) while all the other screens get a
+		 * slug based on the parent screen title (ex: `languages_page_mlang_strings`, where `languages` is the parent
+		 * screen's slug). This will not prevent the `toplevel` hooks to fire, but it will fire the `languages` hooks in
+		 * addition: this way, screens can be removed or moved around without the need of hooking both prefixes: using
+		 * the hooks with the `languages` prefix will work in both cases.
 		 *
-		 * @param array $tabs list of tab names
+		 * @see get_plugin_page_hookname()
 		 */
-		$tabs = apply_filters( 'pll_settings_tabs', $tabs );
-
-		$parent = '';
+		foreach ( array( 'load-', 'admin_print_styles-', 'admin_print_scripts-', 'admin_head-', '', 'admin_print_footer_scripts-', 'admin_footer-' ) as $prefix ) {
+			add_action(
+				"{$prefix}toplevel_page_{$parent}",
+				function () use ( $prefix, $page_type, $parent ) {
```

Comment:
```suggestion
				static function () use ( $prefix, $page_type, $parent ) {
```
Since it's called in a loop, let's make it the less impactful possible.